OSOE-482: Enforce parameter splatting instead of backtick in Lombiq.Analyzers.PowerShell

BenedekFarkas opened the pull request:
[OSOE-482](https://lombiq.atlassian.net/browse/OSOE-482)
Fixes #14

BenedekFarkas committed:
Adding Measure-Backtick community analyzer rule

Copied from https://github.com/PowerShell/PSScriptAnalyzer/blob/master/Tests/Engine/CommunityAnalyzerRules/CommunityAnalyzerRules.psm1#L613

BenedekFarkas committed:
Cleaning up and improving Measure-LineContinuation

BenedekFarkas committed:
Updating tests to detect/expect PSAvoidUsingLineContinuation

BenedekFarkas committed:
Temporarily disabling the execution of the build using the NuGet package

BenedekFarkas committed:
Fixing syntax for Windows PowerShell in Measure-LineContinuation

Piedone commented on .github/workflows/test-analysis-failure.yml:
```diff
@@ -10,6 +10,7 @@ on:
 jobs:
   call-test-analysis-failure-nuget:
     name: Test Analysis Failure - NuGet PackageReference
+    if: ${{ false }}
```
This seems like a leftover. BTW you don't need to use the `${{ ... }}` notation in `if`s.

Piedone commented on Lombiq.Analyzers.PowerShell/Rules/Measure-LineContinuation/Measure-LineContinuation.psm1:
```diff
@@ -0,0 +1,57 @@
+    [Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]]
+.NOTES
+    Copied (and improved) version of
+    https://github.com/PowerShell/PSScriptAnalyzer/blob/master/Tests/Engine/CommunityAnalyzerRules/CommunityAnalyzerRules.psm1#L613.
```
Also include [its license](https://github.com/PowerShell/PSScriptAnalyzer/blob/master/LICENSE#L12).

BenedekFarkas replied:
I added it to `Measure-AutomaticVariableAlias` too just in case, because it started out as a copy of the same module as this one.

Piedone reviewed (changes requested):
Add a "Fixes" reference to https://github.com/Lombiq/PowerShell-Analyzers/issues/14 in the description, please.

BenedekFarkas committed:
Fixing code styling in test-analysis-failure workflow

BenedekFarkas committed:
Fixing indentation

BenedekFarkas committed:
Adding license from https://github.com/PowerShell/PSScriptAnalyzer/blob/master/LICENSE to the custom rules that were derived from CommunityAnalyzerRules

https://github.com/PowerShell/PSScriptAnalyzer/blob/master/Tests/Engine/CommunityAnalyzerRules/CommunityAnalyzerRules.psm1

BenedekFarkas commented:
> Add a "Fixes" reference to #14 in the description, please.

Can we automate that? We know the Jira issue key from the PR, so we can try to find an appropriate issue (since issue titles are suffixed with it by the community activity workflow if it all started with a GH issue) by looking at the issue title suffixes.

BenedekFarkas committed:
Fixing indentation in static-code-analysis workflow

BenedekFarkas committed:
Testing simplified syntax to run analyzers from Windows PowerShell only if the runner OS is Windows

BenedekFarkas committed:
Fixing incorrect copypaste

Piedone commented:
I guess we could like that, yes. Please open an issue for that in the LGHA repo.

Piedone commented on .github/workflows/test-analysis-failure.yml:
```diff
@@ -10,10 +10,10 @@ on:
   call-test-analysis-failure-nuget:
     name: Test Analysis Failure - NuGet PackageReference
-    if: ${{ false }}
+    if: false
```
This still seems like a leftover (we don't need a no-op step).

BenedekFarkas replied:
Yes, fixed! The NuGet reference test solution still needs to be beta tested.
Edit: All green with the beta package.

Piedone commented on .github/workflows/static-code-analysis.yml:
```diff
@@ -7,10 +7,6 @@ on:
     branches:
       - dev
 
-defaults:
```
See my notes about the workflow here: https://github.com/Lombiq/Utility-Scripts/pull/28#discussion_r1052640912 However, do we actually need this in this repo? Since it's already under OSOCE, this analysis should run.

BenedekFarkas replied:
Yes, because OSOCE only runs PS Core analysis, but this workflow runs both Windows PS and PS Core. This makes sure that Invoke-Analyzers.ps1 (and the custom analyzer rules) comply with Windows PS, otherwise consumer repositories won't be able to run analysis on it.

Piedone replied:
I see, so for this repo we actually have to run a superset of analysis happening in OSOCE otherwise.

BenedekFarkas commented:
> I guess we could like that, yes. Please open an issue for that in the LGHA repo.

https://github.com/Lombiq/GitHub-Actions/issues/151

BenedekFarkas committed:
Removing leftover/temporary change

BenedekFarkas committed:
Updating Lombiq.Analyzers.PowerShell.PackageReference.csproj to use the beta package built from the issue branch

BenedekFarkas committed:
Merge branch 'dev' into issue/OSOE-482

# Conflicts:
#	TestSolutions/Lombiq.Analyzers.PowerShell.PackageReference/Lombiq.Analyzers.PowerShell.PackageReference.csproj

BenedekFarkas committed:
Updating static-code-analysis workflow to use the checkout action from LGHA
